### DISCUSSION TITLE
Contract already verified.

### DISCUSSION THREAD

**Discussion by Abelaby** [Cyfrin/foundry-full-course-cu#234] [Q&A] [answered]
I have a weird situation here, In lesson 6, I deployed the SimpleStorage contract to sepolia using sepolia rpc and my private key.
But when I checked the contract, it was already verified.
I redeployed using another wallet, that contract is also verified automatically, might I know why this is happening?

Here are the hashes for both the transactions:

```
0x0ad04525f178d1077b901ec637e3423e3b879654293df0e620dfd38c79f52b22
0xa680d2833d33a5185e7c8ad5008b27d94fe0ce23a6b01a2463341d6a8a58a958

```

**Comment by ghost** [comment, marked as answer]
Hello @Abelaby it is not weird at all, in some cases Etherscan is able to match the bytecode of a newly deployed contract with an already verified contract. It's like an auto verification feature from Etherscan.

Sometimes it works and in other cases is unable to match the bytecode.

**Reply by Abelaby** [reply]
Tysm for clearing that mate
